tests/rptest/services/kgo_verifier_services.py

```diff
@@ -673,11 +673,29 @@ def wait_node(self, node, timeout_sec=None):
 
         return super().wait_node(node, timeout_sec=timeout_sec)
 
-    def wait_for_acks(self, count, timeout_sec, backoff_sec):
-        self._redpanda.wait_until(
-            lambda: self._status_thread.errored or self._status.acked >= count,
-            timeout_sec=timeout_sec,
-            backoff_sec=backoff_sec)
+    def wait_for_acks(self,
+                      count,
+                      timeout_sec,
+                      backoff_sec,
+                      progress_sec=None):
+        def acks():
+            return self._status.acked
+
+        def acks_at_count(fn):
+            return self._status_thread.errored or fn() >= count
+
+        if progress_sec is not None:
+            self._redpanda.wait_until_with_progress_check(
+                acks,
+                acks_at_count,
```

Comment:
does acks_at_count need to be a closure that passes in an `fn` parameter?

Reply:
not really :)

Reply:
oh i didn't have a problem with acks_at_count taking the fn parameter, i just meant that when passed to _redpanda.wait_until_with_progress_check it wouldn't work without being a closure since it did take fn as a parameter?

Reply:
ah, i see. the closure would get constructed under the hood for the underlying `wait_until` call like `lambda: acks_at_count(acks)`. goofy API. felt necessary at the time.